controllers/hyperconverged/hyperconverged_controller.go

```diff
@@ -1045,6 +1055,34 @@ func (r *ReconcileHyperConverged) detectTaintedConfiguration(req *common.HcoRequ
 	}
 }
 
+func (r *ReconcileHyperConverged) getSystemHealthStatus(req *common.HcoRequest) string {
+	isConditionAvailableTrue := req.Conditions.IsStatusConditionTrue(hcov1beta1.ConditionAvailable)
+	isConditionDegradedTrue := req.Conditions.IsStatusConditionTrue(hcov1beta1.ConditionDegraded)
+	isConditionProgressingTrue := req.Conditions.IsStatusConditionTrue(hcov1beta1.ConditionProgressing)
+	isConditionReconcileCompleteTrue := req.Conditions.IsStatusConditionTrue(hcov1beta1.ConditionReconcileComplete)
+
+	isSystemHealthStatusError := !isConditionAvailableTrue || isConditionDegradedTrue
+	if isSystemHealthStatusError {
+		if metricErr := metrics.HcoMetrics.SetHCOMetricSystemHealthStatusError(); metricErr != nil {
+			req.Logger.Error(metricErr, "failed to update the systemHealthStatus metric")
+		}
+		return systemHealthStatusError
+	}
+
+	isSystemHealthStatusWarning := isConditionProgressingTrue || !isConditionReconcileCompleteTrue
+	if isSystemHealthStatusWarning {
+		if metricErr := metrics.HcoMetrics.SetHCOMetricSystemHealthStatusWarning(); metricErr != nil {
+			req.Logger.Error(metricErr, "failed to update the systemHealthStatus metric")
+		}
+		return systemHealthStatusWarning
+	}
+
+	if metricErr := metrics.HcoMetrics.SetHCOMetricSystemHealthStatusHealthy(); metricErr != nil {
+		req.Logger.Error(metricErr, "failed to update the systemHealthStatus metric")
+	}
+	return systemHealthStatusHealthy
+}
+
```

Comment:
Not sure about this function. First, its name: it promises to get the status, but it also sets the metric. In general, function should do one thing. What do you think about splitting it to two functions, one to get the status and one to switch-case the status to set the right metric?